internal/discovery/namespace_filter.go

```diff
@@ -0,0 +1,128 @@
+package discovery
+
+import (
+	"errors"
+
+	"github.com/newrelic/nri-kubernetes/v3/internal/config"
+	"github.com/newrelic/nri-kubernetes/v3/internal/storer"
+
+	log "github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	listersv1 "k8s.io/client-go/listers/core/v1"
+)
+
+// NamespaceFilterer provides an interface to filter namespaces.
+type NamespaceFilterer interface {
+	IsAllowed(namespace string) bool
```

Comment:
Minor naming thing, but I have a weakness for single-word methods :)

```suggestion
	Match(namespace string) bool
```

Reply:
Done in e57ba695a157903790cc9878ad0f6cbf032beb9f.